Form/Type/Filter/ChoiceType.php

```diff
@@ -47,6 +47,10 @@ public function buildForm(FormBuilderInterface $builder, array $options)
             self::TYPE_EQUAL => $this->translator->trans('label_type_equals', array(), 'SonataAdminBundle'),
             self::TYPE_CONTAINS_WORDS => $this->translator->trans('label_type_contains_words', array(), 'SonataDoctrinePHPCRAdmin'),
         );
+        
+        if (!method_exists('Symfony\Component\Form\FormTypeInterface', 'setDefaultOptions')) {
+            $choices = array_flip($choices);
+        }
```

Comment:
It's a start but it should be a bit more complex, with a condition to add the `choices_as_value` for and only for sf 2 8